src/lib/sw_engine/tvgSwRasterNeon.h

```diff
@@ -101,28 +101,28 @@ static inline bool neonRasterTranslucentRect(SwSurface* surface, const SwBBox& r
     uint8x8_t vIalpha = (uint8x8_t) vdup_n_u8((uint8_t) ialpha);
 
     uint8x8_t *vDst = NULL;
+    bool align;
 
     for (uint32_t y = 0; y < h; ++y) {
-        uint32_t align = 0;
         uint32_t *dst = &buffer[y * surface->stride];
 
         if ((((uint32_t) dst) & 0x7) != 0) {
-            vDst = (uint8x8_t*)(dst+1);
-            align = 1;
+            *dst = color + ALPHA_BLEND(*dst, ialpha);
+            dst++;
+            align = true;
         } else {
-            vDst = (uint8x8_t*) dst;
+            align = false;
         }
+        vDst = (uint8x8_t*)dst;
 
-        uint32_t iterations = w / 2;
-        uint32_t left = w % 2;
-
-        if (align) *dst = color + ALPHA_BLEND(*dst, ialpha);
+        uint32_t iterations = (w - align) / 2;
```

Comment:
looks bad  int - boolean.

Reply:
jap, I think that too :D
I wrote that since you did the same in #740 in neonRasterTranslucentRle
Please merge yours when it's ready and then I'll change this one.

Reply:
it's bad showcase. you'd rather -1 for it. (even I didn't know it)